x-pack/plugins/timelines/server/search_strategy/timeline/factory/events/all/index.ts

```diff
@@ -37,7 +37,6 @@ export const timelineEventsAll: TimelineFactory<TimelineEventsQueries.all> = {
   ): Promise<TimelineEventsAllStrategyResponse> => {
     // eslint-disable-next-line prefer-const
     let { fieldRequested, ...queryOptions } = cloneDeep(options);
-    queryOptions.fields = buildFieldsRequest(fieldRequested, queryOptions.excludeEcsData);
```

Comment:
When this line is removed, timeline requests are made with an empty `fields` array, i.e.

```json
  "fields": [],
```

per the screenshot of the JSON diff below, where the left hand side is an inspected timeline request running from the `main` branch, and the right hand side is a request from the PR branch:

![fields_request](https://user-images.githubusercontent.com/4459398/160929688-55136d09-7865-44a8-9a2c-be7ea5aa17b0.png)

_Above: left: a request from `main`, right: a request from the PR branch_

Each of the `fields` entries in the screenshot above on the left have the following form:

```json
    {
      "field": "event.action",
      "include_unmapped": true
    },
```

The `"include_unmapped": true` entry is required to return unmapped fields because, per [this section of the Fields API documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-fields.html#retrieve-unmapped-fields)

> By default, the fields parameter returns only values of mapped fields.

I think more research is required to determine whether or not using a wildcard `*` would eliminate the need to populate the `fields` array with individual field names, but either way, it's likely still a requirement to ensure requests still make use of the `"include_unmapped": true` parameter to the Fields API.

Reply:
> I think more research is required to determine whether or not using a wildcard `*` would eliminate the need to populate the `fields` array with individual field names, but either way, it's likely still a requirement to ensure requests still make use of the `"include_unmapped": true` parameter to the Fields API.

An example request that uses both the wildcard and `"include_unmapped": "true"` would look like this:

```json
  "fields": [
    {
      "field": "*",
      "include_unmapped": "true"
    }
  ],
```

Reply:
Requests currently make use of the [docvalue_fields](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-fields.html#docvalue-fields) API, per the example below:

```json
{
  "docvalue_fields": [
    {
      "field": "@timestamp"
    },
    {
      "field": "agent.build.original"
    },
    // ...
 ]
}
```

Per a conversation with @XavierM, it's no longer necessary to use the `docvalue_fields` to workaround date formatting issues, because they are solved by the (newer) fields API.

Consider:
- removing the `docvalue_fields` API from requests
- in it's place, always including (regardless of the mappings), an entry in the `fields` array for `@timestamp`, shown in the sample request below:

```json
  "fields": [
    {
      "field": "@timestamp",
      "format": "strict_date_optional_time"
    },
    // ...
   ]
```

- populating `fields` will entries for all `date` (and `date_nanos`?) types, per the example below:

```json
  "fields": [
    {
      "field": "*",
      "include_unmapped": "true"
    },
    {
      "field": "@timestamp",
      "format": "strict_date_optional_time"
    },
    {
      "field": "code_signature.timestamp",
      "format": "strict_date_optional_time"
    },
    {
      "field": "dll.code_signature.timestamp",
      "format": "strict_date_optional_time"
    },
    // ...
  ]
```

The example request above:
- For completeness, also includes the wildcard entry for `"field": "*"`
- Always includes an entry for `"field": "@timestamp"`, regardless of the mappings
- Includes entries for additional `date` fields, e.g.  `"field": "code_signature.timestamp"` and `"field": "dll.code_signature.timestamp"`, based on the mappings

If requests follow the example format above, it will no longer be necessary to use the `docvalue_fields` API.

Reply:
Is the omission of

```
"unmapped_type": "date"
```

in the `sort` section of the diff above intentional?

Reply:
> in the sort section of the diff above intentional?

removed the changes